### PR TITLE
Openspace vis_contents

### PR DESCRIPTION
**Pull request by Camper12345**
* Openspaces now use vis_contents as a displaying method
* Floor plane system. Object on different floors now have different planes.
      - It seems like nothing is broken.

### CODE REVIEW

File: maps/CEVEris/_map_data_Eris.dm [hunk 1]
```diff
@@ -203,28 +206,27 @@ ADMIN_VERB_ADD(/client/proc/test_MD, R_DEBUG, null)
 	var/is_accessable_level = TRUE // Prob modifier for random access (space travelling)
 	var/generate_asteroid= FALSE
 	var/tmp/z_level
+	var/height = -1	///< The number of Z-Levels in the map.
```
Comment (LethalGhost) [inline maps/CEVEris/_map_data_Eris.dm]: Why default height is -1?

**Comment by NanakoAC** [comment]
This is going to need a lot more explanation

what exactly is vis contents, why make this change, what are the benefits to us?

File: maps/CEVEris/_map_data_Eris.dm (hunk 1 above)
Comment (Camper12345) [reply maps/CEVEris/_map_data_Eris.dm]: This is for no one forgot to set the height ont he map. It's asserted at the New()

**Comment by LethalGhost** [comment]
@NanakoAC 
`vis_contents` is new feature what was introdused in byond 512.
It's allow objects to be dispayed on another objects. Like overlays but respect dispayed objects changes (transform matrix, icon, dir changes, etc)

That PR replace original open space code (based on overlays for objects and "shadows" for mobs).
Key benefits is:
- Native support instead custom overlays based code
- Smoth movement for objects visible through open space
- Respect to all changes for objects visible through open space
- It's allow extended multi-z interations for players (overlays-system bounds objects to turf, `vis_contents` one - not)


I'm also agree. That PR required careful checks it can couse some bugs with multi-z interactions with objects and mobs.

**Comment by LethalGhost** [comment]
@NanakoAC 
There was lots of disscussion about that in Shipyard last few days.
But.. well... probably mostly in russian.

File: code/__defines/_planes+layers.dm [hunk 2]
```diff
@@ -136,16 +136,16 @@ What is the naming convention for planes or layers?
...
 //HUD layer defines
 
-#define FULLSCREEN_PLANE 18
+#define FULLSCREEN_PLANE 32000
 #define FLASH_LAYER 18
 #define FULLSCREEN_LAYER 18.1
 #define UI_DAMAGE_LAYER 18.2
 
-#define HUD_PLANE 19
+#define HUD_PLANE 32680
```
Comment (NanakoAC) [inline code/__defines/_planes+layers.dm]: why are these numbers so high? Not that it matters, but seems rather arbitrary

File: code/game/objects/items/weapons/tools/tape.dm [hunk 3]
```diff
@@ -126,24 +126,36 @@
...
 /obj/item/weapon/ducttape/proc/attach(var/obj/item/weapon/W)
 	stuck = W
 	W.forceMove(src)
-	if (istype(W, /obj/item/weapon/paper))
-		icon_state = W.icon_state + "_taped"
-		overlays = W.overlays
+	update_icon()
+	name = W.name + " (taped)"
+
+/obj/item/weapon/ducttape/update_icon()
```
Comment (NanakoAC) [inline code/game/objects/items/weapons/tools/tape.dm]: why is duct tape modified here?

File: code/game/turfs/simulated/floor_icon.dm [hunk 4]
```diff
@@ -308,7 +308,7 @@ var/list/flooring_cache = list()
 	if(!flooring_cache[cache_key])
 		var/image/I = image(icon = 'icons/turf/damage_overlays.dmi', icon_state = icon_base)
 
-		I.plane = FLOOR_PLANE
```
Comment (NanakoAC) [inline code/game/turfs/simulated/floor_icon.dm]: when would a floor not be on the floor plane?

File: code/modules/lighting/lighting_overlay.dm [hunk 5]
```diff
@@ -17,6 +17,16 @@
 
 	var/needs_update = FALSE
 
+/*/atom/movable/lighting_overlay/init_plane()
```
Comment (NanakoAC) [inline code/modules/lighting/lighting_overlay.dm]: some commented out code that seems to have been added here?

File: code/modules/mob/inventory.dm [hunk 6]
```diff
@@ -149,7 +149,7 @@
 	if (src.client)
 		src.client.screen -= I
 	I.layer = initial(I.layer)
-	I.plane = initial(I.plane)
+	I.set_plane(initial(I.plane))
```
Comment (NanakoAC) [inline code/modules/mob/inventory.dm]: set_plane seems to only be used in a few cases, while many examples of directly setting the plane remain.
Why so?

File: maps/CEVEris/_map_data_Eris.dm (hunk 1 above)
Comment (NanakoAC) [reply maps/CEVEris/_map_data_Eris.dm]: Are you sure? It looks like it was asserted at new, but you've deleted this ? https://i.imgur.com/cjpEcXw.png

File: code/__defines/_planes+layers.dm (hunk 2 above)
Comment (Camper12345) [reply code/__defines/_planes+layers.dm]: It's close to 32767 which is maximum plane value. No z-level planes should be higher than hud, so this is good way to ensure it

File: code/game/objects/items/weapons/tools/tape.dm (hunk 3 above)
Comment (Camper12345) [reply code/game/objects/items/weapons/tools/tape.dm]: Duct tape allows you to tape items to the walls and uses some plane magic for this

File: code/game/turfs/simulated/floor_icon.dm (hunk 4 above)
Comment (Camper12345) [reply code/game/turfs/simulated/floor_icon.dm]: Floor plane is no longer constant, but decal still needs to be be on the same plane as the floor

File: code/modules/mob/inventory.dm (hunk 6 above)
Comment (Camper12345) [reply code/modules/mob/inventory.dm]: There is not so many places where plane is assigned, excluding object definitions. I changed almost all of them.

**Pull request merged**